Add Skills to Characters for 5e engine

nazar opened the pull request:
References #168

nazar committed:
#168 - wip

nazar committed:
Merge branch 'master' into 168_skillz

nazar committed:
closes #168 - Add Skills to Characters for 5e engine

mikelax commented:
@nazar  this is looking really nice! I have one small comment on the UI for how each Skill component should be rendered. I think it should be something like the following. 
The textfield can be made much smaller to create space as it will only hold a single digit. 
The change is to make it obvious to the user that the toggle element applies to Proficiency. 

SkillName (Ability)     Proficient (toggle element)   textfield
ex: 
Acrobatics (Dex)    Proficient (toggle)   textfield


Also, it appears that all the elements are top-aligned within the Skill Component. Can we make this middle aligned?

nazar committed:
#168 - various

1. rearrange skill items
2. grid cell content alignment
3. auto calc skill modifiers when creating a character and not when editing an existing char

nazar committed:
#168 - cleanup validators

nazar committed:
#168 - tweak skill labels

nazar committed:
#168 - fix autocalc for abilities on edit but not for skills

mikelax reviewed (approved):
🚀
